### ISSUE TITLE
poetry env info not detecting local active python version

### ISSUE THREAD

**Issue by petobens** [python-poetry/poetry#1870]
- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**:  Arch Linux
- **Poetry version**:  1.0.2
- **pyproject.toml file**: 
```toml
[tool.poetry]
name = "poetry-pyenv"
version = "0.1.0"
description = ""
authors = ["John Doe <jon@doe.com>"]

[tool.poetry.dependencies]
python = "~3.7"

[tool.poetry.dev-dependencies]

[build-system]
requires = ["poetry>=0.12"]
build-backend = "poetry.masonry.api"
```

## Issue

Start with a directory `poetry-pyenv` that only contains the `pyproject.toml` file above and, assuming pyenv is already installed, do:

1. Ensure that we are using python 3.7 (previously installed with `pyenv install 3.7.5`):
```commandline
> echo "3.7.5" > .python-version | cat .python-version
3.7.5
> pyenv versions
  system
* 3.7.5 (set by /home/pedro/Desktop/poetry-pyenv/.python-version)
> python --version
3.7.5
```
2. Run `poetry shell` so as to create a new virtualenv
```commandline
> poetry shell
The currently activated Python version 3.8.1 is not supported by the project (~3.7).
Trying to find and use a compatible version.
Using python3 (3.7.5)
Creating virtualenv poetry-pyenv-3OBOihHY-py3.7 in /home/pedro/.cache/pypoetry/virtualenvs
Spawning shell within /home/pedro/.cache/pypoetry/virtualenvs/poetry-pyenv-3OBOihHY-py3.7
. /home/pedro/.cache/pypoetry/virtualenvs/poetry-pyenv-3OBOihHY-py3.7/bin/activate
```
3. Exit the above subshell (by pressing `Ctrl-d` for example`) and check the virtualenv info
```commandline
> poetry env info
Virtualenv
Python:         3.8.1
Implementation: CPython
Path:           NA

System
Platform: linux
OS:       posix
Python:   /home/pedro/.local/pipx/venvs/poetry
```
4.  Check that the 3.7 virtualenv actually exists and its unique:
```commandline
> poetry env list 
poetry-pyenv-3OBOihHY-py3.7
```

Basically after 2. I would've expect `poetry env info` to correctly report that the only active virtualenv is the one corresponding to python 3.7. However it reports that there is no active virtual environment under python 3.8.1 which is my global (system) python but not my currently local/pyenv active python (which is indeed python 3.7.5). If I run `poetry env use 3.7` then `poetry env info` does indeed correctly report an active venv under python 3.7.5 however I would've expect this to happen automatically (i.e without needing the user to actually run `poetry env use`).

**Comment by Italosayan** [comment]
Having the same issue

**Comment by finswimmer** [comment]
Hello,

this issue is quite old and lots of improvements take place. Setting [virtualenvs.prefer-active-python](https://python-poetry.org/docs/configuration/#virtualenvsprefer-active-python-experimental) to `true` should solve the problem.

Feel free to leave a comment if it doesn't work for you.

fin swimmer
